plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -52,7 +68,95 @@ export default function Content(): JSX.Element {
 		} ) );
 	};
 
+	const loadMoreProducts = useCallback( () => {
+		setIsLoadingMore( true );
+		const params = new URLSearchParams();
+		const abortController = new AbortController();
+
+		if ( query.category && query.category !== '_all' ) {
+			params.append( 'category', query.category );
+		}
+
+		if ( query.term ) {
+			params.append( 'term', query.term );
+		}
+
+		const wccomSettings = getAdminSetting( 'wccomHelper', false );
+		if ( wccomSettings.storeCountry ) {
+			params.append( 'country', wccomSettings.storeCountry );
+		}
+
+		params.append( 'page', ( currentPage + 1 ).toString() );
+
+		fetchSearchResults( params, abortController.signal )
+			.then( ( productList ) => {
+				setAllProducts( ( prevProducts ) => {
+					const flattenedPrevProducts = Array.isArray(
+						prevProducts[ 0 ]
+					)
+						? prevProducts.flat()
+						: prevProducts;
+
+					const newProducts = productList.products.filter(
+						( newProduct ) =>
+							! flattenedPrevProducts.some(
+								( prevProduct ) =>
+									prevProduct.id === newProduct.id
+							)
+					);
+
+					if ( newProducts.length > 0 ) {
+						setFirstNewProductId( newProducts[ 0 ].id ?? 0 );
+					}
+
+					const combinedProducts = [
+						...flattenedPrevProducts,
+						...newProducts,
+					];
+
+					setSearchResultsCount( {
+						extensions: combinedProducts.filter(
+							( p ) => p.type === ProductType.extension
+						).length,
+						themes: combinedProducts.filter(
+							( p ) => p.type === ProductType.theme
+						).length,
+						'business-services': combinedProducts.filter(
+							( p ) => p.type === ProductType.businessService
+						).length,
+					} );
+
+					return combinedProducts;
+				} );
+
+				speak( __( 'More products loaded', 'woocommerce' ) );
+				setCurrentPage( ( prevPage ) => prevPage + 1 );
+				setIsLoadingMore( false );
+			} )
+			.catch( () => {
+				speak( __( 'Error loading more products', 'woocommerce' ) );
+			} )
+			.finally( () => {
+				setIsLoadingMore( false );
+			} );
+
+		return () => {
+			abortController.abort();
+		};
+	}, [
+		currentPage,
+		query.category,
+		query.term,
+		setIsLoadingMore,
+		setSearchResultsCount,
+	] );
+
 	useEffect( () => {
+		// if it's a paginated request, don't use this effect
+		if ( currentPage > 1 ) {
+			return;
+		}
```

Comment:
I see the value of `currentPage` is only affected within `loadMoreProducts` - we may want to reset it when switching tabs?

Reply:
Scratch this, I see we're doing it [here](https://github.com/woocommerce/woocommerce/pull/51434/files#diff-66d84dd7feb9389b0808515aed92c9eebc6bad572bd192d6ca327bee3b6796d1R343).

Reply:
This is taken care of by the useEffect hook on line 344